### DISCUSSION TITLE
Welcome to langchain-chatbot Discussions!

### DISCUSSION THREAD

**Discussion by Haste171** [Haste171/langchain-chatbot#4] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by tristan-mcinnis** [comment]
Thanks for pulling this together. I've already gotten a lot further than the original one shared by Mayo.

**Reply by Haste171** [reply]
Glad to hear this repo was helpful to you!

**Reply by tristan-mcinnis** [reply]
I got it working last night! Just a quick question: after 'ingesting' docs - would it be wise to completely create a new index if I want to query a new set of documents? I realized I got responses from 'older' docs - that had been ingested - even after re-launching streamlit from terminal. Think that makes sense since Pinecone and what I was working on are separate.

**Reply by ahgsql** [reply]
The namespace got responsible for this.You need to make it dynamic or change manually for creating another namespace
on  streamlit.py there is **pinecone_namespace** on line 21

**Comment by tristan-mcinnis** [comment]
If you use the 'streamlit' version, why is it that you always have to re-insert your different API, Index? 

from langchain.document_loaders import DirectoryLoader, PyMuPDFLoader
from langchain.text_splitter import RecursiveCharacterTextSplitter
from langchain.embeddings.openai import OpenAIEmbeddings
from langchain.chains import ConversationalRetrievalChain
from langchain.chat_models import ChatOpenAI
from langchain.vectorstores import Pinecone
import pinecone
from templates.qa_prompt import QA_PROMPT
from templates.condense_prompt import CONDENSE_PROMPT
from dotenv import load_dotenv
load_dotenv()

pinecone_namespace ='testing-pdf-2389203901'

def main():
    # Set Streamlit app title and header
  
I've already updated the '.env' with the relevant details.

**Reply by Haste171** [reply]
There is no current database to hold your inputs. There is a website interface in development to allow more features though.
